fsm.go

```diff
@@ -255,6 +257,9 @@ func (f *FSM) Cannot(event string) bool {
 // The last error should never occur in this situation and is a sign of an
 // internal bug.
 func (f *FSM) Event(event string, args ...interface{}) error {
+	f.eventMutex.Lock()
```

Comment:
I think I can see the intent, but doesn't this just let Event() behave like with the other mutex but before the unlocks in the callbacks were added? I have a feeling that all this would be better performed using a channel instead of mutexes, but I don't have a suggestetion at the moment. What do you think? Any ideas that spring to mind.